### PR TITLE
2019-12-30 - post java-jersey-jaxrs

### PR DESCRIPTION
**Pull request by bdemers**
This blog post has the following features:

- [x] A GitHub Repository with a polished README
- [x] A title that's approved by Dev Marketing
- [ ] A URL that's approved by Dev Marketing
- [x] Rendered locally and confirmed that no Markdown typos exist

### CODE REVIEW

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 1]
```diff
@@ -0,0 +1,595 @@
+---
+layout: blog_post
+title: "Build a Secure REST Application Using Jersey"
+author: daniel-pereira
+description: "Tutorial: Learn how to build a secure REST application with JAX-RS and Jersey"
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Description needs a period.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+
+The first step to create your REST application using https://eclipse-ee4j.github.io/jersey/[Jersey] is to create the project structure.
+
+You'll use https://start.spring.io/[Spring Initializer] to create the application.
+
+Go to `https://start.spring.io/` and fill in the following information:
+
+* Project: Maven Project
+* Language: Java
+* Group: com.okta
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]:
```suggestion
* Group: `com.okta`
```

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+* Artifact: jersey-rest
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]:
```suggestion
* Artifact: `jersey-rest`
```

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 4]
```diff
@@ -0,0 +1,595 @@
...
+
+@Configuration // <.>
+public class JerseyConfiguration extends ResourceConfig { // <.>
+
+    public JerseyConfiguration() {
+        packages("com.okta.jerseyrest.resource"); // <.>
+    }
+}
+----
+
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Remove unnecessary line break.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 5]
```diff
@@ -0,0 +1,595 @@
...
+   }'
+----
+
+Replace `<task_id>` with the ID of one of the tasks you created previously.
+
+Great job! If you list your tasks again you'll see that the description changed.
+
+You implemented all the CRUD functions, except for the last one. Let's finish it by implementing the delete endpoint.
+
+== Delete a Entry
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Delete a Task might be better.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+* Log in to your account
+* Click in **Applications**
+* Click on **Add Application**. You will be redirected to the following page:
+
+image::{% asset_path 'blog/java-jersey-jaxrs/okta-applications.png' %}[alt=Okta application page,width=600,align=center]
+
+* Select **Web** and click **Next**
+* Fill in the following options in the form:
+** Name: my-first-app
+** Base URIs: http://localhost:8080
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]:
```suggestion
** Base URIs: `http://localhost:8080`
```

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 7]
```diff
@@ -0,0 +1,595 @@
...
+[source,java]
+----
+package com.okta.jerseyrest.configuration;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: This annotation is not necessary. It's included in [`EnableWebSecurity`](https://docs.spring.io/spring-security/site/docs/4.0.x/apidocs/org/springframework/security/config/annotation/web/configuration/EnableWebSecurity.html).

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 8]
```diff
@@ -0,0 +1,595 @@
...
+X-Content-Type-Options: nosniff
+X-XSS-Protection: 1; mode=block
+Cache-Control: no-cache, no-store, max-age=0, must-revalidate
+Pragma: no-cache
+Expires: 0
+X-Frame-Options: DENY
+Content-Type: text/html;charset=utf-8
+Content-Language: en
+Content-Length: 802
+Date: Mon, 04 Nov 2019 12:52:52 GMT
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: I'd change this date so it's close to publish date.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 9]
```diff
@@ -0,0 +1,595 @@
...
+== Generate a Valid Token
+
+To validate your request you need to add the `Authorization` header to the request. The header will provide the type of authentication and the token, which will look like the snippet below:
+
+[source,text]
+----
+-H 'Authorization: Bearer <token>'
+----
+
+To generate the token you can access https://oidcdebugger.com/ and provide the following information:
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]:
```suggestion
To generate the token you can access <https://oidcdebugger.com> and provide the following information:
```

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 10]
```diff
@@ -0,0 +1,595 @@
...
+
+- Authorize URI: `\https://{yourOktaDomain}/oauth2/default/v1/authorize`
+- Redirect URI: `\https://oidcdebugger.com/debug`
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Remove backslash from the above two links.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 11]
```diff
@@ -0,0 +1,595 @@
...
+< HTTP/1.1 200
+< X-Content-Type-Options: nosniff
+< X-XSS-Protection: 1; mode=block
+< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
+< Pragma: no-cache
+< Expires: 0
+< X-Frame-Options: DENY
+< Content-Type: application/json
+< Content-Length: 2
+< Date: Thu, 21 Nov 2019 10:15:36 GMT
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Change date to be closer to pub date.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 12]
```diff
@@ -0,0 +1,595 @@
...
+It now returns the task you just created!
+
+[source,json]
+----
+[{"id":"a44dba4f-d239-441a-925d-d9248aeb4925","description":"Test my Jersey App!"}]
+----
+
+Well done! You managed to create a CRUD service using Jersey! Even better, the service is secure and it took you minimal effort to make it happen.
+
+You can view the source code of this tutorial going to its https://github.com/oktadeveloper/jersey-rest-example[GitHub repository].
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: Repo is missing the standard `okta-` prefix. Not sure it matters, but this is a convention we've been using for a while now. I drop the prefix when the example is not secured by Okta.

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc [hunk 13]
```diff
@@ -0,0 +1,595 @@
...
+
+== Learn More!
```
Comment (mraible) [inline _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: I'm surprised this h2 made it past editorial. It usually says something like "Learn More about `<keywords>`".

Comment (bdemers) [reply _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: It changed a couple of times, it was too long and wrapped, but I'll change it to something like "Learn More About Jersey and REST!"

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc (hunk 10 above)
Comment (bdemers) [reply _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: I ran into an odd asciidoc issue, that caused an html tag escape problem, adding the backslashes stopped the href rendering, and the broken tag issue. Have you ever seen that?

File: _source/_posts/2019-12-30-java-jersey-jaxrs.adoc (hunk 5 above)
Comment (bdemers) [reply _source/_posts/2019-12-30-java-jersey-jaxrs.adoc]: "Entry" matches the others headers, but i did fix the "a" -> "an" issue

**Pull request merged**